evalml/pipelines/components/transformers/imputers/per_column_imputer.py

```diff
@@ -18,6 +18,10 @@ class PerColumnImputer(Transformer):
         default_impute_strategy (str): Impute strategy to fall back on when none is provided for a certain column.
             Valid values include "mean", "median", "most_frequent", "constant" for numerical data,
             and "most_frequent", "constant" for object data types. Defaults to "most_frequent".
+        impute_all (bool): Whether or not to impute all columns or just the columns that are specified in `impute_strategies`. If True,
+            all columns will be imputed either using the strategy specified in `impute_strategies` or using the `default_impute_strategy`.
+            If False, only columns specified as keys in the `impute_strategies` dictionary are imputed. If False and `impute_strategies` is None,
+            no columns will be imputed. Defaults to True.
```

Comment:
Would it be worth it to warn the user in the case that the imputer does not impute anything?

Reply:
Added a warning during fit time for this case :)